core/jobs/batch_jobs/exploration_migration_jobs.py

```diff
@@ -0,0 +1,503 @@
+        ) % (exp_domain.Exploration.CURRENT_EXP_SCHEMA_VERSION)
+        change_dicts = [change.to_dict() for change in exploration_changes]
+        with datastore_services.get_ndb_context():
+            models_to_put = updated_exploration_model.compute_models_to_commit(
+                feconf.MIGRATION_BOT_USERNAME,
+                feconf.COMMIT_TYPE_EDIT,
+                commit_message,
+                change_dicts,
+                additional_models={'rights_model': exp_rights_model}
+            ).values()
+        datastore_services.update_timestamps_multi(list(models_to_put))
+        return models_to_put
+
+    @staticmethod
+    def _update_exploration_summary(
+        migrated_exploration: exp_domain.Exploration,
+        exp_summary: exp_domain.ExplorationSummary,
+        exp_summary_model: exp_models.ExpSummaryModel,
+        exp_rights: exp_models.ExplorationRightsModel
+    ) -> exp_models.ExpSummaryModel:
+        """Generates newly updated exploration summary model.
+
+        Args:
+            migrated_exploration: Exploration. The migrated exploration
+                domain object.
+            exp_summary: ExplorationSummary. The exploration
+                summary domain object.
+            exp_summary_model: ExpSummaryModel. The exploration
+                summary model to update.
+            exp_rights: ExplorationRightsModel. The exploration
+                rights model.
+
+        Returns:
+            ExpSummaryModel. The updated exploration summary model to put into
+            the datastore.
+        """
+        if exp_summary_model is not None:
+            old_exp_summary = exp_fetchers.get_exploration_summary_from_model(
+                exp_summary_model)
+            ratings = old_exp_summary.ratings or feconf.get_empty_ratings()
+            scaled_average_rating = (
+                exp_services.get_scaled_average_rating(ratings))
+        else:
+            ratings = feconf.get_empty_ratings()
+            scaled_average_rating = feconf.EMPTY_SCALED_AVERAGE_RATING
+
+        contributors_summary = (
+            exp_summary_model.contributors_summary if exp_summary_model else {})
+        contributor_ids = list(contributors_summary.keys())
+
+        last_human_update_ms = 0
+
+        current_version = migrated_exploration.version
+        version_nums = list(range(1, current_version + 1))
+
+        with datastore_services.get_ndb_context():
+            snapshots_metadata = (
+                exp_models.ExplorationModel.get_snapshots_metadata(
+                    migrated_exploration.id,
+                    version_nums,
+                    allow_deleted=False
+                )
+            )
+
+        for snapshot_metadata in reversed(snapshots_metadata):
+            if snapshot_metadata['committer_id'] != feconf.MIGRATION_BOT_USER_ID: # pylint: disable=line-too-long
+                last_human_update_ms = snapshot_metadata['created_on_ms']
+                break
```

Comment:
I don't think we need this part, as this is only a migration and we don't expect the `last_human_update_ms` to be updated anyway.